### DISCUSSION TITLE
Form values not saving

### DISCUSSION THREAD

**Discussion by Cnote43** [Budibase/budibase#10965] [Help] [answered]
I built a very simple app about a year ago using a budibase database. As a learning exercise I'm trying to recreate the app using an MySQL database. I created a form and defined an action for the Save button to Save Row. The Save button will create a new row, but it's blank except for the Date (added as a defined action on the Save button) and the row ID columns. 

In my previously built app, I could select the Form as a Datasource when setting up the Save Row action on the Save button. I noticed when setting up the Save Row action on the Save button for this new app that I can't pick anything for the Datasource, except None. Is this why the form data won't save? If so, how do I get the form to show up as a data source?

Edited for clarity.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Cnote43,

Could you provide screenshots of your setup?

- In your data area do you have different data sources setup/ have tables available?
- Is your form nested inside a data provider component?
- Is your save row action attached to the data provider thats the parent of your form?

Thanks ahead of time!

**Reply by Cnote43** [reply]
Your questions helped this noob figure it out. Thank you! I didn't have a data provider as a parent of my form. In my old app I didn't have that either but it worked. Maybe because I built it in an earlier version?

After adding a data provider as a parent to the form I was then able to select the form as the datasource in the Save button's action.

**Reply by ConorWebb96** [reply]
Glad to have helped!

I'm not 100% sure on this, but I think they have always existed? Could you possibly have used the `Form Block` instead of the `Form` component?

The form block does a lot of the configuration for you, making it so you only have to select the table you wish to target.
